sympy/core/frac.py

```diff
@@ -0,0 +1,345 @@
+from collections import defaultdict
+from collections import defaultdict
+from functools import cmp_to_key
+
+import sympy
+from sympy.core.numbers import Integer
+from .basic import Basic
+from .cache import cacheit
+from .expr import Expr
+from .logic import _fuzzy_group
+from .operations import AssocOp
+from .singleton import S
+
+
+# internal marker to indicate:
+#   "there are still non-commutative objects -- don't forget to process them"
+class NC_Marker:
+    is_Order = False
+    is_Frac = False
+    is_Number = False
+    is_Poly = False
+
+    is_commutative = False
+
+
+# Key for sorting commutative args in canonical order
+_args_sortkey = cmp_to_key(Basic.compare)
+
+
+def _fracsort(args):
+    # in-place sorting of args
+    args.sort(key=_args_sortkey)
+
+
+def _unevaluated_Frac(*args):
+    args = list(args)
+    newargs = []
+    ncargs = []
+    co = S.One
+    while args:
+        a = args.pop()
+        if a.is_Frac:
+            c, nc = a.args_cnc()
+            args.extend(c)
+            if nc:
+                ncargs.append(Frac._from_args(nc))
+        elif a.is_Number:
+            co *= a
+        else:
+            newargs.append(a)
+    _fracsort(newargs)
+    if co is not S.One:
+        newargs.insert(0, co)
+    if ncargs:
+        newargs.append(Frac._from_args(ncargs))
+    return Frac._from_args(newargs)
+
+
+class Frac(Expr, AssocOp):
+    __slots__ = ()
+
+    is_Frac = True
+
+    def __neg__(self):
+        return sympy.Mul(self.args[0], sympy.Pow(self.args[1], -1, evaluate=False), evaluate=False).__neg__
```

Comment:
what about a `def to_mul()` and wrap `sympy.Mul(self.args[0], sympy.Pow(self.args[1], -1, evaluate=False)`. There are 31 occurrences in this class.